internal/provider/resource_gitlab_group.go

```diff
@@ -284,6 +321,10 @@ func resourceGitlabGroupRead(ctx context.Context, d *schema.ResourceData, meta i
 	d.Set("runners_token", group.RunnersToken)
 	d.Set("share_with_group_lock", group.ShareWithGroupLock)
 	d.Set("default_branch_protection", group.DefaultBranchProtection)
+	d.Set("membership_lock", group.MembershipLock)
```

Comment:
I'm not seeing `membership_lock` coming back when running a GET on a group, so I'm a bit confused where this would be coming from.

Reply:
You know, I think you're right~ I just took another read through their Group API documentation, and it doesn't list this attribute as one that's returned. It's kinda vague on _exactly_ which attributes are returned other than providing an example, but their example doesn't explicitly mentioned it and it's not mentioned as an "additional" attribute when licensed at a higher level either. Really odd that you can pass it in both create/update, but it doesn't come back in "get", but that exactly described the behavior I'm seeing in the test.

Nice catch @jtymes !

Reply:
I agree that it's strange. It _seems_ like it should be fairly easy for them to expose. I'll take a look and maybe make an upstream MR.

Reply:
Yes, this happens .. I've had a similar encounter yesterday : https://gitlab.com/gitlab-org/gitlab/-/issues/354657